### PR TITLE
Resource Management Implementation

### PR DESCRIPTION
**Pull request by mattmess1221**
[API](https://github.com/SpongePowered/SpongeAPI/pull/2135)

Implement basic resource management
TODO: init events and pack finders

### CODE REVIEW

File: src/main/java/org/spongepowered/common/resource/meta/SpongeMetadataSectionSerializer.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import com.google.gson.JsonParseException;
+import net.minecraft.resources.data.IMetadataSectionSerializer;
+import org.spongepowered.api.data.persistence.DataFormats;
+import org.spongepowered.api.data.persistence.DataView;
+import org.spongepowered.api.resource.meta.MetaSection;
+
+import java.io.IOException;
+
+/**
+ * A fallback metadata section serializer to get a specific section without a
```
Comment (mattmess1221) [inline src/main/java/org/spongepowered/common/resource/meta/SpongeMetadataSectionSerializer.java:38]: This doc is outdated. It was written for a previous version of the class, which had a different purpose.

File: src/main/java/org/spongepowered/common/resource/pack/SpongeFileSystemPack.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    public Collection<ResourcePath> find(PackType type, String prefix, int depth, Predicate<String> filter) {
+        try {
+            Path root = getPackRoot(type);
+            return getNamespaces(type).stream()
+                    .flatMap(namespace -> {
+                        try {
+                            return Files.walk(root.resolve(namespace).resolve(prefix), depth)
+                                    .filter(s -> !s.getFileName().toString().endsWith(".mcmeta"))
+                                    // todo is the path relative to start?
+                                    .map(Object::toString)
```
Comment (mattmess1221) [inline src/main/java/org/spongepowered/common/resource/pack/SpongeFileSystemPack.java]: I need to do some testing on this to see if the path is relative. If it is not, I will have to relativize it manually before converting it to a string.

**Comment by mattmess1221** [comment]
I believe I've completed the implementation. I will mark this PR as ready for review as soon as I can test it.

**Comment by Zidane** [comment]
@killjoy1221 

We still plan to bring this into api-8. Do you have some time lately to update it?

**Comment by Zidane** [comment]
This has been brought into the implementation via a different approach.

Regardless, thank you for all your work on this killjoy.

**Pull request closed without merging**